budgetportal/webflow/views.py

```diff
@@ -164,3 +250,26 @@ class ProvInfraProjectSearchView(FacetMixin, HaystackViewSet):
         "status_order",
         "estimated_completion_date",
     ]
+
+    def list(self, request, *args, **kwargs):
+        csv_download_params = self._get_csv_query_params(request.query_params)
+        response = super().list(request, *args, **kwargs)
+        response.data["csv_download_url"] = "{}?{}".format(
+            reverse("provincial-infrastructure-project-api-csv"), csv_download_params
+        )
+        return response
+
+    @action(detail=False, methods=["get"])
+    def get_csv(self, request, *args, **kwargs):
+        self.serializer_class = self.csv_serializer_class
+        return super().list(request, *args, **kwargs)
+
+    def _get_csv_query_params(self, original_query_params):
+        csv_download_params = original_query_params.copy()
+        csv_download_params.pop("fields", None)
+        csv_download_params.pop("ordering", None)
+        csv_download_params.pop("limit", None)
+        params = ""
+        for param_name, param_value in csv_download_params.items():
+            params += "{}={}".format(param_name, param_value)
```

Comment:
please use urllib.parse.urlencode

Reply:
I meant for the `"{}={}"` you pass in a dictionary and it does the key=value

Reply:
instead of doing `"{}={}"``and `&` yourself, it takes care of all of that

Reply:
right, it fits here well.